engine/src/flutter/shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java

```diff
@@ -1230,8 +1230,124 @@ public void setTextInputEditingState_nullInputMethodSubtype() {
+            TextInputPlugin textInputPlugin =
+                new TextInputPlugin(
+                    testView,
+                    textInputChannel,
+                    scribeChannel,
+                    mock(PlatformViewsController.class),
+                    mock(PlatformViewsController2.class));
+            textInputPlugin.setTextInputClient(
+                0,
+                new TextInputChannel.Configuration(
+                    false,
+                    false,
+                    true,
+                    true,
+                    false,
+                    TextInputChannel.TextCapitalization.NONE,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null,
+                    null));
+            ImeSyncDeferringInsetsCallback imeSyncCallback = textInputPlugin.getImeSyncCallback();
+            FlutterEngine flutterEngine =
+                spy(new FlutterEngine(ctx, mockFlutterLoader, mockFlutterJni));
+            FlutterRenderer flutterRenderer = spy(new FlutterRenderer(mockFlutterJni));
+            when(flutterEngine.getRenderer()).thenReturn(flutterRenderer);
+            assertEquals(0, testImm.getRestartCount(testView));
+            // FlutterView restarts the input method to inform the Android framework that an engine
+            // has
+            // been attached.
+            testView.attachToFlutterEngine(flutterEngine);
+            assertEquals(1, testImm.getRestartCount(testView));
+
+            ArgumentCaptor<FlutterRenderer.ViewportMetrics> viewportMetricsCaptor =
+                ArgumentCaptor.forClass(FlutterRenderer.ViewportMetrics.class);
+
+            WindowInsets.Builder builder = new WindowInsets.Builder();
+
+            // There's a pending restart since we initialized the text input client. Flush that now.
+            textInputPlugin.setTextInputEditingState(
+                testView, new TextInputChannel.TextEditState("", 0, 0, -1, -1));
+            assertEquals(2, testImm.getRestartCount(testView));
+
+            // Set the initial insets and verify that they were set and the bottom view inset is
+            // correct.
+            builder.setInsets(WindowInsets.Type.ime(), Insets.of(0, 0, 0, 0));
+            when(testView.getRootWindowInsets()).thenReturn(builder.build());
+            imeSyncCallback.getInsetsListener().onApplyWindowInsets(testView, builder.build());
+            verify(flutterRenderer, atLeast(1)).setViewportMetrics(viewportMetricsCaptor.capture());
+            assertEquals(0, viewportMetricsCaptor.getValue().viewInsetBottom);
+
+            // Imm restarts when clearTextInputClient is called while the IME is hidden.
+            textInputPlugin.clearTextInputClient();
+            assertEquals(3, testImm.getRestartCount(testView));
+          });
+    }
+  }
+
   @Test
   public void destroy_clearTextInputMethodHandler() {
+
```

Comment:
This new line looks unintentional? (Feel free to skip if this is intentional!)

```suggestion
```

Reply:
Thanks for catching that!